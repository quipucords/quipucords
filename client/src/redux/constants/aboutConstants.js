const ABOUT_DIALOG_OPEN = 'ABOUT_DIALOG_OPEN';
const ABOUT_DIALOG_CLOSE = 'ABOUT_DIALOG_CLOSE';

export { ABOUT_DIALOG_OPEN, ABOUT_DIALOG_CLOSE };
