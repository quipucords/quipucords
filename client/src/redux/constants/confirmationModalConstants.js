const CONFIRMATION_MODAL_SHOW = 'CONFIRMATION_MODAL_SHOW';
const CONFIRMATION_MODAL_HIDE = 'CONFIRMATION_MODAL_HIDE';

export { CONFIRMATION_MODAL_SHOW, CONFIRMATION_MODAL_HIDE };
