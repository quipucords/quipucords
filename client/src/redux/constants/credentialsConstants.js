
const LOAD_CREDENTIALS_SUCCESS = 'LOAD_CREDENTIALS_SUCCESS';

export {
  LOAD_CREDENTIALS_SUCCESS
}
