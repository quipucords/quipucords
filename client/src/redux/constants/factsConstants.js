const ADD_FACTS_ERROR = 'ADD_FACTS_ERROR';
const ADD_FACTS_LOADING = 'ADD_FACTS_LOADING';
const ADD_FACTS_SUCCESS = 'ADD_FACTS_SUCCESS';

export { ADD_FACTS_ERROR, ADD_FACTS_LOADING, ADD_FACTS_SUCCESS };
