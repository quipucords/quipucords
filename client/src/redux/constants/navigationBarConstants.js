const NAV_TOGGLE_COLLAPSE = 'NAV_TOGGLE_COLLAPSE';
const NAV_COLLAPSE = 'NAV_COLLAPSE';
const NAV_OPEN = 'NAV_OPEN';

export { NAV_TOGGLE_COLLAPSE, NAV_COLLAPSE, NAV_OPEN };
