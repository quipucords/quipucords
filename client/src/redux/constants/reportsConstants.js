const GET_REPORT = 'GET_REPORT';
const GET_MERGE_REPORT = 'GET_MERGE_REPORT';

export { GET_REPORT, GET_MERGE_REPORT };
