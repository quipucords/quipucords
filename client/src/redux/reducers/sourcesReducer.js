import _ from 'lodash';
import helpers from '../../common/helpers';
import { sourcesTypes } from '../constants';

const initialState = {
  view: {
    error: false,
    errorMessage: '',
    pending: false,
    fulfilled: false,
    sources: []
  },

  update: {
    error: false,
    errorMessage: '',
    pending: false,
    fulfilled: false,
    sourceId: '',
    delete: false
  }
};

const sourcesReducer = function(state = initialState, action) {
  switch (action.type) {
    // Error/Rejected
    case helpers.REJECTED_ACTION(sourcesTypes.DELETE_SOURCE):
    case helpers.REJECTED_ACTION(sourcesTypes.DELETE_SOURCES):
      return helpers.setStateProp(
        'update',
        {
          error: action.error,
          errorMessage: helpers.getErrorMessageFromResults(action.payload),
          delete: true
        },
        {
          state,
          initialState
        }
      );

    case helpers.PENDING_ACTION(sourcesTypes.DELETE_SOURCE):
    case helpers.PENDING_ACTION(sourcesTypes.DELETE_SOURCES):
      return helpers.setStateProp(
        'update',
        {
          pending: true,
          delete: true
        },
        {
          state,
          initialState
        }
      );

    case helpers.FULFILLED_ACTION(sourcesTypes.DELETE_SOURCE):
    case helpers.FULFILLED_ACTION(sourcesTypes.DELETE_SOURCES):
      return helpers.setStateProp(
        'update',
        {
          fulfilled: true,
          delete: true
        },
        {
          state,
          initialState
        }
      );

    case helpers.REJECTED_ACTION(sourcesTypes.GET_SOURCES):
      return helpers.setStateProp(
        'view',
        {
          error: action.error,
          errorMessage: helpers.getErrorMessageFromResults(action.payload)
        },
        {
          state,
          initialState
        }
      );

    // Loading/Pending
    case helpers.PENDING_ACTION(sourcesTypes.GET_SOURCES):
      return helpers.setStateProp(
        'view',
        {
          pending: true,
          sources: state.view.sources
        },
        {
          state,
          initialState
        }
      );

    // Success/Fulfilled
    case helpers.FULFILLED_ACTION(sourcesTypes.GET_SOURCES):
      // Get resulting sources and update the selected state of each
      const sources = _.get(action, 'payload.data.results', []);
      return helpers.setStateProp(
        'view',
        {
          sources: sources,
          fulfilled: true
        },
        {
          state,
          initialState
        }
      );

    default:
      return state;
  }
};

sourcesReducer.initialState = initialState;

export { initialState, sourcesReducer };

export default sourcesReducer;
