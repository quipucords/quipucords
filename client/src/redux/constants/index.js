import * as aboutTypes from './aboutConstants';
import * as credentialsTypes from './credentialsConstants';
import * as factsTypes from './factsConstants';
import * as navigationBarTypes from './navigationBarConstants';
import * as reportsTypes from './reportsConstants';
import * as scansTypes from './scansConstants';
import * as sourcesTypes from './sourcesConstants';
import * as viewToolbarTypes from './viewToolbarConstants';

const reduxTypes = {
  about: aboutTypes,
  credentials: credentialsTypes,
  facts: factsTypes,
  navigation: navigationBarTypes,
  reports: reportsTypes,
  scans: scansTypes,
  sources: sourcesTypes,
  viewToolbar: viewToolbarTypes
};

export {
  reduxTypes,
  aboutTypes,
  credentialsTypes,
  factsTypes,
  navigationBarTypes,
  reportsTypes,
  scansTypes,
  sourcesTypes,
  viewToolbarTypes
};

export default reduxTypes;
