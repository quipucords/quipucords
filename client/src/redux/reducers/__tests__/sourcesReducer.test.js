import helpers from '../../../common/helpers';
import { sourcesTypes } from '../../constants/index';
import sourcesReducer from '../sourcesReducer';

const initialState = {
  view: {
    error: false,
    errorMessage: '',
    pending: false,
    fulfilled: false,
    sources: []
  },

  update: {
    error: false,
    errorMessage: '',
    pending: false,
    fulfilled: false,
    sourceId: '',
    delete: false
  }
};

describe('SourcesReducer', () => {
  it('should return the initial state', () => {
    expect(sourcesReducer(undefined, {})).toEqual(initialState);
  });

  it('should handle DELETE_SOURCE_REJECTED', () => {
    const dispatched = {
      type: helpers.REJECTED_ACTION(sourcesTypes.DELETE_SOURCE),
      error: true,
      payload: {
        message: 'BACKUP MESSAGE',
        response: {
          data: {
            detail: 'DELETE ERROR'
          }
        }
      }
    };

    const resultState = sourcesReducer(undefined, dispatched);

    expect(resultState.update.error).toBeTruthy();
    expect(resultState.update.errorMessage).toEqual('DELETE ERROR');
    expect(resultState.update.delete).toBeTruthy();

    expect(resultState.view).toEqual(initialState.view);
  });

  it('should handle GET_SOURCES_REJECTED', () => {
    const dispatched = {
      type: helpers.REJECTED_ACTION(sourcesTypes.GET_SOURCES),
      error: true,
      payload: {
        message: 'BACKUP MESSAGE',
        response: {
          data: {
            detail: 'GET ERROR'
          }
        }
      }
    };

    const resultState = sourcesReducer(undefined, dispatched);

    expect(resultState.view.error).toBeTruthy();
    expect(resultState.view.errorMessage).toEqual('GET ERROR');

    expect(resultState.update).toEqual(initialState.update);
  });

  it('should handle GET_SOURCES_PENDING', () => {
    const dispatched = {
      type: helpers.PENDING_ACTION(sourcesTypes.GET_SOURCES)
    };

    const resultState = sourcesReducer(undefined, dispatched);

    expect(resultState.view.pending).toBeTruthy();

    expect(resultState.update).toEqual(initialState.update);
  });

  it('should handle GET_SOURCES_FULFILLED', () => {
    const dispatched = {
      type: helpers.FULFILLED_ACTION(sourcesTypes.GET_SOURCES),
      payload: {
        data: {
          results: [
            {
              name: '1',
              id: 1
            },
            {
              name: '2',
              id: 2
            },
            {
              name: '3',
              id: 3
            },
            {
              name: '4',
              id: 4
            }
          ]
        }
      }
    };

    const resultState = sourcesReducer(undefined, dispatched);

    expect(resultState.view.fulfilled).toBeTruthy();
    expect(resultState.view.sources).toHaveLength(4);

    expect(resultState.update).toEqual(initialState.update);
  });
});
