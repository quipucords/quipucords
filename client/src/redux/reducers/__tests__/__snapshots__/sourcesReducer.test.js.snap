// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fulfilled types DELETE_SOURCE 1`] = `
Object {
  "result": Object {
    "update": Object {
      "delete": true,
      "error": false,
      "errorMessage": "",
      "fulfilled": true,
      "pending": false,
      "sourceId": "",
    },
    "view": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "sources": Array [],
      "updateSources": false,
    },
  },
  "type": "DELETE_SOURCE_FULFILLED",
}
`;

exports[`fulfilled types DELETES_SOURCE 1`] = `
Object {
  "result": Object {
    "update": Object {
      "delete": true,
      "error": false,
      "errorMessage": "",
      "fulfilled": true,
      "pending": false,
      "sourceId": "",
    },
    "view": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "sources": Array [],
      "updateSources": false,
    },
  },
  "type": "DELETES_SOURCE_FULFILLED",
}
`;

exports[`fulfilled types GET_SOURCES 1`] = `
Object {
  "result": Object {
    "update": Object {
      "delete": false,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "sourceId": "",
    },
    "view": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": true,
      "pending": false,
      "sources": Array [],
      "updateSources": false,
    },
  },
  "type": "GET_SOURCES_FULFILLED",
}
`;

exports[`pending types DELETE_SOURCE 1`] = `
Object {
  "result": Object {
    "update": Object {
      "delete": true,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": true,
      "sourceId": "",
    },
    "view": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "sources": Array [],
      "updateSources": false,
    },
  },
  "type": "DELETE_SOURCE_PENDING",
}
`;

exports[`pending types DELETES_SOURCE 1`] = `
Object {
  "result": Object {
    "update": Object {
      "delete": true,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": true,
      "sourceId": "",
    },
    "view": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "sources": Array [],
      "updateSources": false,
    },
  },
  "type": "DELETES_SOURCE_PENDING",
}
`;

exports[`pending types GET_SOURCES 1`] = `
Object {
  "result": Object {
    "update": Object {
      "delete": false,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "sourceId": "",
    },
    "view": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": true,
      "sources": Array [],
      "updateSources": false,
    },
  },
  "type": "GET_SOURCES_PENDING",
}
`;

exports[`rejected types DELETE_SOURCE 1`] = `
Object {
  "result": Object {
    "update": Object {
      "delete": true,
      "error": true,
      "errorMessage": "ERROR",
      "fulfilled": false,
      "pending": false,
      "sourceId": "",
    },
    "view": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "sources": Array [],
      "updateSources": false,
    },
  },
  "type": "DELETE_SOURCE_REJECTED",
}
`;

exports[`rejected types DELETES_SOURCE 1`] = `
Object {
  "result": Object {
    "update": Object {
      "delete": true,
      "error": true,
      "errorMessage": "ERROR",
      "fulfilled": false,
      "pending": false,
      "sourceId": "",
    },
    "view": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "sources": Array [],
      "updateSources": false,
    },
  },
  "type": "DELETES_SOURCE_REJECTED",
}
`;

exports[`rejected types GET_SOURCES 1`] = `
Object {
  "result": Object {
    "update": Object {
      "delete": false,
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "sourceId": "",
    },
    "view": Object {
      "error": true,
      "errorMessage": "ERROR",
      "fulfilled": false,
      "pending": false,
      "sources": Array [],
      "updateSources": false,
    },
  },
  "type": "GET_SOURCES_REJECTED",
}
`;
